wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/Utility/MessageToolboxDataSource.swift

```diff
@@ -190,8 +193,32 @@ class MessageToolboxDataSource {
 
     /// Returns the status for the sender of the message.
     fileprivate func selfStatus(for message: ZMConversationMessage) -> NSAttributedString? {
-        guard let sender = message.senderUser,
-              sender.isSelfUser else { return nil }
+        guard let sender = message.senderUser, sender.isSelfUser else {
+            return nil
+        }
+
+        var deliveryStateString: String
+
+        switch message.deliveryState {
+        case .pending:
+            deliveryStateString = ContentSystem.pendingMessageTimestamp
+        case .read:
+            return selfStatusForReadDeliveryState(for: message)
+        case .delivered:
+            deliveryStateString = ContentSystem.messageDeliveredTimestamp
+        case .sent:
+            deliveryStateString = ContentSystem.messageSentTimestamp
+        case .invalid, .failedToSend:
+            return nil
+        }
+
+        return NSAttributedString(string: deliveryStateString) && attributes
+    }
```

Comment:
There's a bit of duplication here that would be good to factor out.

Reply:
Done!